File: LinkedList/PYTHON/RotateLinkedList.py
from email import header


class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def insert(self, data):
        newNode = Node(data)
        if self.head == None:
            self.head = newNode
        else:
            ptr1 = self.head
            while (ptr1.next != None):
                ptr1 = ptr1.next
            ptr1.next = newNode

    def display(self):
        ptr1 = self.head
        while (ptr1 != None):
            print(ptr1.data, end="->")
            ptr1 = ptr1.next
        print(None)

    def rotate(self, k):
        length = 1
        ptr1 = self.head
        while (ptr1.next != None):
            ptr1 = ptr1.next
            length += 1
        print(ptr1.data)
        print(length)
        k=k%length
        print(k)
        ptr1.next=self.head
        newHead=ptr1
        i=0
        while (i<k):
            


obj = LinkedList()
obj.insert(1)
obj.insert(2)
obj.insert(3)
obj.insert(4)
obj.insert(5)
obj.display()
obj.rotate(8)

File: LinkedList/JAVA/RotateLinkedList.java
// public class RotateLinkedList {
//     class Node {
//         int data;
//         Node next;

//         public Node(int val) {
//             data = val;

//         }
//     }

//     Node head;

//     public RotateLinkedList() {
//         head = null;
//     }

//     public void addElement(int data) {
//         Node newNode = new Node(data);
//         if (head == null) {
//             head = newNode;
//         } else {
//             Node temp = head;
//             while (temp.next != null) {
//                 temp = temp.next;
//             }
//             temp.next = newNode;
//         }
//     }

//     public void display(Node head) {
//         Node temp = head;
//         while (temp != null) {
//             System.out.print(temp.data + "->");
//             temp = temp.next;
//         }
//         System.out.println("null");
//     }

//     public Node rotateList(Node head, int k) {
//         Node temp = head;
//         if (head == null || head.next == null) {
//             return head;
//         }
//         int length = 1;
//         while (temp.next != null) {

//             length++;

//             temp = temp.next;
//         }
//         k = k % length;
//         k = length - k;
//         temp.next = head;
//         System.out.println("temp.next ="+temp.next.data);
//         for (int j = k; j > 0; j--) {
//             temp = temp.next;
//         }
//         System.out.println("temp.next ="+temp.next.data);
//         head = temp.next;
//         System.out.println("newHead ="+head.data);
//         temp.next = null;
//         return head;

//     }

//     public static void main(String[] args) {
//         RotateLinkedList obj = new RotateLinkedList();
//         obj.addElement(1);
//         obj.addElement(2);
//         obj.addElement(3);
//         obj.addElement(4);
//         obj.addElement(5);
//         obj.display(obj.head);
//         Node newHead = obj.rotateList(obj.head, 2);
//         obj.display(newHead);
//     }
// }
